In [3]:
import pandas as pd
import os

In [5]:
data_list = os.listdir('/nfs/masi/MCL/nifti/combine')
f = open('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/subject0825.txt', 'w')

for subj in data_list:
    f.write( subj + '\n')
f.close()

f = open('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/sess0825.txt', 'w')

for subj in data_list:
    sess_list = os.listdir('/nfs/masi/MCL/nifti/combine/' + subj )
    for sess in sess_list:
        f.write( subj + 'time' + sess +  '\n')
f.close()


In [29]:
df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword.csv')
df = df.loc[df['MCL_ID'] == df['MCL_ID']]
print (len(set(df['MCL_ID'])))
print (len(set(data_list)))
print (len(set(df['MCL_ID']) | set(data_list) ))


1199
2283
2545


In [23]:
subj_diag = {}

df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0421/Report_part1_nodup.csv')

for i, item in df.iterrows():
    if item['cancer_bengin'] not in [0, 1]:
        continue
    subj = str(item['MCL_ID']).replace('.0','')
    if subj in subj_diag.keys():
        subj_diag[subj] = max(subj_diag[subj], item['cancer_bengin'])
    else:
        subj_diag[subj] = item['cancer_bengin']

df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword.csv')
for i, item in df.iterrows():
    if item['gt_reg'] not in [0, 1]:
        continue
    subj = str(item['MCL_ID']).replace('.0','')
    subj_diag[subj] = item['gt_reg']
    


In [24]:
 len(set(data_list) & set(subj_diag.keys()))

1431

In [17]:
subj_mayo = {}

df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword.csv')
for i, item in df.iterrows():
    if item['gt_reg'] not in [0, 1]:
        continue
    if item['Mayo.Risk'] != item['Mayo.Risk']:
        continue
    subj = str(item['MCL_ID']).replace('.0','')
    subj_mayo[subj] = item['Mayo.Risk']
print (len(set(data_list) & set(subj_mayo.keys())) )
    

771


In [33]:
subj_blood = {}

df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword.csv')
for i, item in df.iterrows():
    if item['gt_reg'] not in [0, 1]:
        continue
    if item['CSI..CYFRA21.1..ng.mL.'] != item['CSI..CYFRA21.1..ng.mL.']:
        continue
    subj = str(item['MCL_ID']).replace('.0','')
    subj_blood[subj] = item['CSI..CYFRA21.1..ng.mL.']
print (len(set(data_list) & set(subj_blood.keys())) )
print (len(set(data_list) & set(subj_blood.keys()) & set(subj_mayo.keys()))) 

739
739


In [37]:
df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword_radi.csv')
print (len(df))
df = df.loc[df['ANTPOST_LENGTH_END_MM_X'] == df['ANTPOST_LENGTH_END_MM_X']]
print (len(df))
df = df.loc[df['Mayo.Risk'] == df['Mayo.Risk']]
print (len(df))
df = df.loc[df['CSI..CYFRA21.1..ng.mL.'] == df['CSI..CYFRA21.1..ng.mL.']]


1199
739
739
661


In [42]:
df = pd.read_csv('/media/gaor2/8e7f6ccf-3585-4815-856e-80ce8754c5b5/data/MCL/file/clinical/LabelFile/0720MK/20200720_nopassword.csv')

df_upmc = df.loc[df['Cohort.1.Vandy.2.UPMC.3.DECAMP.4.UCD'] == '3']
print (len(df_upmc))


146


# statisical 20200825

| Total Subject | 2545 | 
| With Image (xnat)   | 2283 |
| With Clinical | 1193 |
| With Diagnosis | 1431 |

| With Mayo & image & Diag | 771 |
| With Mayo & blood & Diag | 739 |
| With Mayo & blood & Mayo & Diag | 739 |

| with Radiomics | 739 |
| with Radiomics & Diagnosis & Mayo_risk | 739 | 
| with Radiomics & Diagnosis & Mayo & blood | 661 | 

|UPMC| 158 |
|DECAMP| 146 |


# the follow for 20201102 

In [ ]:
df = pd.read_csv('')